### ISSUE TITLE
`@stylistic/unit-case` recases the text of an interpolation that holds whitespace

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#298]
Found while reviewing the branch that closes [#234](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/234). Reproduces on `main` before that branch as it does after it.

### What minimal example or steps are needed to reproduce the bug?

A unit standing in front of an interpolation whose text holds whitespace:

```scss
a { b: 10px#{$a != $b}; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-scss",
  "rules": {
    "@stylistic/unit-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

The unit in capitals and the interpolation untouched, `a { b: 10PX#{$a != $b}; }` — or nothing reported at all, the rule leaving a value it cannot read alone, as it does for `a { b: 10px#{$a}; }`.

### What actually happened?

```scss
a { b: 10PX#{$A != $b}; }
```

The name of the variable is recased with the unit. Sass names are case-sensitive, so a stylesheet that compiled stops compiling, and the warning says so out loud:

```
1:10-1:16  Expected "px$a" to be "PX$A"
```

`px$a` is no unit, and no user can act on that message.

### What is the cause?

`isStandardSyntaxValue` asks `hasInterpolation`, whose patterns want a closing brace: `SCSS_INTERPOLATION` is `/#\{.+?\}/su` and `TPL_INTERPOLATION` is `/\{.+?\}/su`. `postcss-value-parser` splits a value at whitespace, so an interpolation holding a space arrives as several word nodes and the first of them, `10px#{$a`, carries no closing brace. The guard never fires, `blurInterpolation` finds nothing to blur, and `valueParser.unit` reads `px$a` as the unit of `10`.

An interpolation with no whitespace in it, `a { b: 10px#{$a!=$b}; }` or `a { b: 10px#{$a}; }`, is one word and is passed over correctly. Less goes the same way: `a { b: 10px@{a b}; }`.

### Do you have a proposal to fix the bug?

Ask whether a word stands inside an interpolation of the value rather than whether that word holds a whole one. The value is at hand, and its interpolations can be found in it once and each word measured against them, the way [findInlineCommentSpans](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/utils/findInlineCommentSpans/index.js) is used for comments.
